internal/hostport/hostport_manager.go

```diff
@@ -238,8 +238,9 @@ func (hm *hostportManager) Remove(id string, podPortMapping *PodPortMapping) (er
 	}
 
 	// exit if there is nothing to remove
+	// don´t forget to clean up opened pod host ports
 	if len(existingChainsToRemove) == 0 {
-		return nil
+		return hm.closeHostports(hostportMappings)
 	}
 
```

Comment:
There's a `return(err)` at line https://github.com/cri-o/cri-o/pull/4542/files#diff-1e8c8253b73c319cc3d39e1e6eebdbff23129076dfe7814b74754f640f68b1c5R261 and I think we can leave behind the hostportmappings there too.  Would it be better to create a defer function to close them there and to do the unlock too?

Reply:
If there is an error programming the iptables rules, it will be possible that the iptables rules for the portmaps are still active so we should keep holding the ports and not close them.